src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx

```diff
@@ -735,21 +734,22 @@ export default class WorkloadPodLogs extends React.Component<WorkloadPodLogsProp
   };
 
   private getContainerInfo = (pod: Pod): ContainerInfo => {
-    const containers = pod.containers ? pod.containers : [];
+    let containers = pod.containers ? pod.containers : [];
     containers.push(...(pod.istioContainers ? pod.istioContainers : []));
-    const containerNames: string[] = containers.map(c => c.name);
-    const options: string[] = [];
-
-    containerNames.forEach(c => {
-      // ignore the proxy
-      if (c !== 'istio-proxy') {
-        if (pod.appLabel && pod.labels) {
-          options[c] = c + '-' + pod.labels['version'];
-        } else {
-          options[c] = c;
-        }
-      }
+    containers = containers.filter(c => c.name !== 'istio-proxy');
+    const options: { [key: string]: string } = {};
+
+    if (containers.length === 0) {
+      options[NoAppContainer] = NoAppContainer;
+      return { container: NoAppContainer, containerOptions: options };
+    }
+
+    const containerNames = containers.map(c => c.name);
+    containerNames.map(n => {
```

Comment:
Nitpick: Shouldn't this be a reduce? At least a `forEach`.

Reply:
yes, I changed it to forEach.